test/data_base.cpp

```diff
@@ -63,150 +116,45 @@ static void test_block_exists(const data_base& interface, size_t height,
-                ////const auto address = prevout.metadata.cache.addresses();
-
-                // for (const payment_address& address: addresses)
-                // {
-                //     auto history = address_store.get(address.hash());
-                //     auto found = false;
-
-                //     for (const payment_record& row: history)
-                //     {
-                //         if (row.hash() == tx_hash && row.index() == j)
-                //         {
-                //             BOOST_REQUIRE_EQUAL(row.height(), height);
-                //             found = true;
-                //             break;
-                //         }
-                //     }
-
-                //     BOOST_REQUIRE(found);
-                // }
-            }
-        }
+        BOOST_REQUIRE_EQUAL(result_tx.position(), tx_position);
 
         if (!catalog)
             return;
 
-        for (size_t j = 0; j < tx.outputs().size(); ++j)
+        if (!tx.is_coinbase())
         {
-            const auto& output = tx.outputs()[j];
-            output_point outpoint{ tx_hash, static_cast<uint32_t>(j) };
-            const auto addresses = output.addresses();
-
-            // for (const payment_address& address: addresses)
-            // {
-            //     auto history = address_store.get(address.hash());
-            //     auto found = false;
-               
-            //     for (const payment_record& row: history)
-            //     {
-            //         BOOST_REQUIRE(row.is_valid());
-
-            //         if (row.hash() == tx_hash && row.index() == j)
-            //         {
-            //             BOOST_REQUIRE_EQUAL(row.height(), height);
-            //             BOOST_REQUIRE_EQUAL(row.data(), output.value());
-            //             found = true;
-            //             break;
-            //         }
-            //     }
-
-            //     BOOST_REQUIRE(found);
-            // }
+            test_inputs_cataloged(payments_store, tx, true);
         }
+
+        test_outputs_cataloged(payments_store, tx, true);
+        ++tx_position;
     }
 }
 
 static void test_block_not_exists(const data_base& interface,
-    const block& block0, bool index_addresses)
+    const block& block, bool catalog)
 {
-    const auto& address_store = interface.addresses();
+    const auto& payments_store = interface.addresses();
 
     // Popped blocks still exist in the block hash table, but not confirmed.
-    const auto block_hash = block0.hash();
+    const auto block_hash = block.hash();
     const auto result = interface.blocks().get(block_hash);
     BOOST_REQUIRE(!is_confirmed(result.state()));
 
-    for (size_t i = 0; i < block0.transactions().size(); ++i)
+    uint32_t tx_position = 0u;
+
+    for (const auto& tx: block.transactions())
     {
-        const auto& tx = block0.transactions()[i];
-        const auto tx_hash = tx.hash();
+        if (!catalog)
```

Comment:
Why is this in the loop, it’s value cannot change in the loop?

Reply:
Moved.